apiserver/facades/client/application/application.go

```diff
@@ -492,13 +492,18 @@ func splitApplicationAndCharmConfigFromYAML(modelType state.ModelType, inYaml, a
 	outYaml string,
 	_ error,
 ) {
-	var allSettings map[string]map[string]interface{}
+	var allSettings map[string]interface{}
 	if err := goyaml.Unmarshal([]byte(inYaml), &allSettings); err != nil {
 		return nil, "", errors.Annotate(err, "cannot parse settings data")
 	}
-	settings, ok := allSettings[appName]
+	settings, ok := allSettings[appName].(map[interface{}]interface{})
 	if !ok {
-		return nil, "", errors.Errorf("no settings found for %q", appName)
+		// Application key not present; it might be 'juju get' output.
+		if _, err := charmConfigFromGet(allSettings); err != nil {
```

Comment:
I'm not sure "FromGet" is the most accurate description.
Maybe "FromConfigValues"?

Reply:
This was the original name (charmConfigFromGetYaml). I will rename it as suggested to make it clear what it does.

Reply:
This and the incorrect comment entries have been addressed via https://github.com/juju/juju/pull/11916/commits/1055b3d5d2ceb86446c3299f0dc04bce6ef30b66